src/v/cloud_topics/state_accessors.h

```diff
@@ -9,22 +9,52 @@
  */
 #pragma once
 
+#include "cloud_topics/level_one/common/file_io.h"
+
+#include <seastar/core/sharded.hh>
+
+#include <memory>
+
+namespace cluster {
+class metadata_cache;
+}
+
 namespace cloud_topics {
 
 class data_plane_api;
 
+namespace l1 {
+class frontend;
+} // namespace l1
+
 // Encapsulates the required bits to access topic state from cloud topics,
 // with minimal dependencies. This allows it to be passed around through
 // different layers without introducing circular dependencies.
 //
 class state_accessors {
 public:
-    explicit state_accessors(data_plane_api* data_plane)
-      : data_plane(data_plane) {}
+    explicit state_accessors(
+      data_plane_api* data_plane,
+      l1::frontend* l1_metastore_fe,
+      l1::file_io* l1_io,
+      cluster::metadata_cache* metadata_cache)
+      : data_plane(data_plane)
+      , l1_metastore_fe(l1_metastore_fe)
+      , l1_io(l1_io)
+      , metadata_cache(metadata_cache) {}
 
     data_plane_api* get_data_plane() { return data_plane; }
 
+    l1::frontend* get_l1_metastore_frontend() { return l1_metastore_fe; }
+
+    l1::file_io* get_l1_io() { return l1_io; }
+
+    cluster::metadata_cache* get_metadata_cache() { return metadata_cache; }
+
 private:
     data_plane_api* data_plane;
+    l1::frontend* l1_metastore_fe;
```

Comment:
Hmm frontend is kind of a bad name for it, it should probably be "RPC router" or something...

In any case any thoughts on adding the metastore here instead of the frontend? I kind of think of `l1::frontend` as an implementation detail of `replicated_metastore`, in which case we're better off using the metastore directly